File: yandex/test_1/test_C.py
import sys
n, m = list(map(int, input().split()))

better = 0
score_v = [0] * n

score_d = {0: n-1}


for game in range(m):
    a, b = list(map(int, sys.stdin.readline().split()))
    diff = a-b
    p = list(map(int, sys.stdin.readline().split()))

    if diff != 0:
        for i in range(10):
            if i == 5:
                diff = -diff
            if p[i] == 0:
                if diff > 0:
                    for j in range(diff):
                        score_v[0] += 1
                        better -= score_d.get(score_v[0], 0)
                else:
                    for j in range(-diff):
                        better += score_d.get(score_v[0], 0)
                        score_v[0] -= 1
            else:
                old_score = score_v[p[i]]
                score_v[p[i]] += diff
                new_score = score_v[p[i]]

                score_d[old_score] -= 1
                score_d[new_score] = 1 + score_d.get(new_score, 0)

                if old_score <= score_v[0] and new_score > score_v[0]:
                    better += 1
                elif old_score > score_v[0] and new_score <= score_v[0]:
                    better -= 1

    print(better)

File: yandex/test_1/test_C.cpp
#include <iostream>

#include <vector>
#include <set>
#include <map>

using namespace std;

typedef long long ll;

int main()
{
    int n, m;
    int g_a, g_b;
    cin >> n >> m;

    vector<int> score_v(n, 0);
    // multiset<int> score_s(score_v.begin(), score_v.end());
    map<int, int> score_m;
    score_m[0] = n;
    vector<int> p(10);
    int better = 0;

    for (int game = 0; game < m; game++)
    {

        cin >> g_a >> g_b;
        int diff = g_a - g_b;
        for (int i = 0; i < 10; i++)
        {
            cin >> p[i];
        }

        if (diff != 0)
        {
            for (int i = 0; i < 10; i++)
            {
                if (i == 5)
                {
                    diff *= -1;
                }
                if (p[i] == 0)
                {
                    if (diff > 0)
                    {
                        for (int j = 1; j <= diff; j++)
                        {
                            better -= score_m[++score_v[0]];
                        }
                    }
                    else
                    {
                        for (int j = 0; j < -diff; j++)
                        {
                            better += score_m[score_v[0]--];
                        }
                    }
                }
                else
                {
                    int old_score = score_v[p[i]];
                    int new_score = score_v[p[i]] += diff;

                    score_m[old_score]--;
                    score_m[new_score]++;

                    if (old_score <= score_v[0] && new_score > score_v[0])
                    {
                        better++;
                    }
                    else if (old_score > score_v[0] && new_score <= score_v[0])
                    {
                        better--;
                    }
                }
            }
        }

        cout << better << endl;
    }

    return 0;
}